Additional test for one-tailed permutation t-test

jCalderTravis opened the pull request:
#### Reference issue
Provides a test relevant for #11511


#### What does this implement/fix?
An existing code test compared two-tailed parametric and non-parametric t-tests. This pull request adds the corresponding code tests for one-tailed t-tests.

larsoner commented on mne/stats/tests/test_permutations.py:
```diff
@@ -58,10 +58,14 @@ def test_permutation_t_test():
-    t_obs_scipy, p_values_scipy = stats.ttest_1samp(X[:, 0], 0)
-    assert_allclose(t_obs[0], t_obs_scipy, 8)
-    assert_allclose(p_values[0], p_values_scipy, rtol=1e-2)
+    tail_codes = {'two-sided': 0, 'less': -1, 'greater': 1}
+    for this_tail, this_code in tail_codes.items():
+        t_obs, p_values, H0 = permutation_t_test(X, n_permutations='all',
+                                                 tail=this_code)
+        t_obs_scipy, p_values_scipy = stats.ttest_1samp(X[:, 0], 0,
+                                                        alternative=this_tail)
+        assert_allclose(t_obs[0], t_obs_scipy, 8)
+        assert_allclose(p_values[0], p_values_scipy, rtol=1e-2)
```
Splitting it this way will make it clearer which tails work and fail rather than stopping at the first failure
```suggestion
@pytest.mark.parametrize('tail', [0, -1, 1])
def test_permutation_t_test_tail(tail):
    """Test that tails work properly."""
    t_obs, p_values, H0 = permutation_t_test(X, n_permutations='all',
                                             tail=this_code)
    t_obs_scipy, p_values_scipy = stats.ttest_1samp(X[:, 0], 0,
                                                    alternative=this_tail)
    assert_allclose(t_obs[0], t_obs_scipy, 8)
    assert_allclose(p_values[0], p_values_scipy, rtol=1e-2)
```

larsoner reviewed (commented):
Nicely replicates the failure!
```
___________________________ test_permutation_t_test ____________________________
mne/stats/tests/test_permutations.py:68: in test_permutation_t_test
    assert_allclose(p_values[0], p_values_scipy, rtol=1e-2)
/usr/share/miniconda3/envs/test/lib/python3.8/contextlib.py:75: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference: 0.35283719
E   Max relative difference: 1.01524173
E    x: array(0.700377)
E    y: array(0.34754)
```
Just pushing a little commit that will help show which tail is failing (maybe both one-sided?)

larsoner commented:
Oh, GitHub won't let me push the commit... feel free to make the suggested change and push @jCalderTravis !

jCalderTravis commented:
Thanks @larsoner for your suggestion. I have added this in, and now we get a test for each tail:

```
FAILED mne/stats/tests/test_permutations.py::test_permutation_t_test_tail[less--1] - AssertionError:
FAILED mne/stats/tests/test_permutations.py::test_permutation_t_test_tail[greater-1] - AssertionError:
```

jCalderTravis commented:
@larsoner would you be happy to merge this test? Thank you.

larsoner commented:
You need to run `black` on your file to get the style checks to be happy

If one of them is expected to fail you'll need to mark it as `xfail` (we never want red / failing tests merged into `main`). So you can do something like:
```
@pytest.mark.parametrize('whatever', [
    x,
    y,
    pytest.param(z, marks=pytest.mark.xfail(reason='Bug in permutations'),
])
```

jCalderTravis commented:
Thanks for your comments @larsoner. I incorporated `xfail` as you suggested. I checked the style with Flake8, and now none of the lines that I added have any style issues.

jCalderTravis commented:
Also now formatted with `black`.

larsoner commented:
Thanks @jCalderTravis !

larsoner commented:
Whoops I didn't realize this was your first PR to MNE-Python! Could you open another quick PR to add this change to `doc/changes/latest.inc` (maybe under the BUG section) with the `:newcontrib:` role along with name/url in `doc/changes/names.inc`? Want to make sure you get credit for this
